pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -98,12 +130,16 @@ func createService(componentMeta metav1.ObjectMeta, componentExt *v1beta1.Compon
 			}
 		} else {
 			port, _ := strconv.Atoi(constants.InferenceServiceDefaultHttpPort)
+			portInt32, err := utils.ConvertStringToInt32(port, 8080)
+			if err != nil {
+				log.Error(err, "Failed to convert port int to int32, using the default value(8080) for port.")
+			}
 			servicePorts = append(servicePorts, corev1.ServicePort{
 				Name: componentMeta.Name,
 				Port: constants.CommonDefaultHttpPort,
 				TargetPort: intstr.IntOrString{
 					Type:   intstr.Int,
-					IntVal: int32(port), // #nosec G109
+					IntVal: portInt32, // #nosec G109
```

Comment:
btw, isn't it better to use the k8s type `IntOrStr` from `apimachinery`?

Reply:
I didn't know the type. Could you please explain what part would be better to use the IntOrStr?

Reply:
updated

Reply:
Line 133 can be removed as well, and instead of setting the intVal, you can set the StrVal, which would save one cast :)

Reply:
good catch!

Reply:
Other tests and source code rely on IntVal, so changing it might cause failures. I’ve decided to keep it as is since it doesn’t seem to be a major issue.

Reply:
we can improve it on an upcoming PR. :)

Reply:
I updated this part based on @israel-hdez recommendation(https://github.com/kserve/kserve/pull/3972#discussion_r1796200467)

Reply:
can the `#nosec G109` be removed now?

Reply:
ok I will close this out